src/state/pools/hooks.ts

```diff
@@ -60,31 +60,83 @@ export const usePools = (): { pools: DeserializedPool[]; userDataLoaded: boolean
     pools: state.pools.data,
     userDataLoaded: state.pools.userDataLoaded,
   }))
-  return { pools: pools.map(transformPool), userDataLoaded }
+  const poolsString = JSON.stringify(pools)
+  const poolsWithUserDataLoading = useMemo(() => {
+    const serializedPools: SerializedPool[] = JSON.parse(poolsString)
+    return { pools: serializedPools.map(transformPool), userDataLoaded }
+  }, [poolsString, userDataLoaded])
+  return poolsWithUserDataLoading
 }
 
 export const usePool = (sousId: number): { pool: DeserializedPool; userDataLoaded: boolean } => {
   const { pool, userDataLoaded } = useSelector((state: State) => ({
     pool: state.pools.data.find((p) => p.sousId === sousId),
     userDataLoaded: state.pools.userDataLoaded,
   }))
-  return { pool: transformPool(pool), userDataLoaded }
+  const poolString = JSON.stringify(pool)
+  const poolWithUserDataLoading = useMemo(() => {
+    const serializedPool: SerializedPool = JSON.parse(poolString)
+    return { pool: transformPool(serializedPool), userDataLoaded }
+  }, [poolString, userDataLoaded])
+  return poolWithUserDataLoading
 }
 
-export const useFetchCakeVault = () => {
+export const usePoolsWithVault = () => {
```

Comment:
Could you use `createSelector` for this case ?

Reply:
Done

Reply:
Done